index.js

```diff
@@ -856,6 +1007,54 @@ converters.ArrayTypeAnnotation = (path, context) /*: K.ArrayType*/ => {
   };
 };
 
+converters.TSIntersectionType = (path, context) /*: K.Intersection*/ => {
+  const types = path.get('types').map(type => convert(type, context));
+  return { kind: 'intersection', types }
+}
+
+converters.TSIndexSignature = (path, context) /*: K.Property */ =>{
+  const id = path.get('parameters')[0];
+  return {
+    kind: 'property',
+    key: {
+      kind: 'id',
+      name: `[${convert(id, context).name}: ${convert(id.get('typeAnnotation'), context).kind}]`
+    },
+    value: convert(path.get('typeAnnotation'), context),
+  };
+};
+
+converters.ImportDeclaration = (path, context) /*: K.Any */ => {
+  return { kind: 'any' };
+};
+
+converters.TSParenthesizedType = (path, context) => {
+  return convert(path.get('typeAnnotation'), context);
+};
+
+converters.TSObjectKeyword = (path, context) /*: K.Obj */ => {
+  return { kind: 'object', members: [] };
+};
+
+converters.TSNullKeyword = (path, context) /*: K.Null */ => {
+  return { kind: 'null' };
+};
+
+converters.TSThisType = (path, context) /*:K.This */ => {
+  return { kind: 'custom', value: 'this' }
+}
+
+function extendedTypesMembers(path, context) {
+  const members = path.get('extends');
+  if (!members || !members.length) {
+    return [];
+  }
+
+  return members
+    .map(e => convert(e, context))
+    .reduce((acc, current) => acc.concat(current.members), []); 
+}
+
 function importConverterGeneral(path, context) /*: K.Import */ {
```

Comment:
TODO: Move the importConverterGeneral and its related functions into its own file (maybe even folder)
- really go to town trying to make this easier to understand and reason about.